flink-ml-lib/src/main/java/org/apache/flink/ml/classification/linear/LogisticRegressionModelData.java

```diff
@@ -0,0 +1,104 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.classification.linear;
+
+import org.apache.flink.api.common.serialization.Encoder;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.connector.file.src.reader.SimpleStreamFormat;
+import org.apache.flink.core.fs.FSDataInputStream;
+import org.apache.flink.core.memory.DataInputViewStreamWrapper;
+import org.apache.flink.core.memory.DataOutputViewStreamWrapper;
+import org.apache.flink.ml.linalg.DenseVector;
+import org.apache.flink.ml.linalg.typeinfo.DenseVectorSerializer;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.table.api.internal.TableImpl;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.OutputStream;
+
+/** Model data of {@link LogisticRegressionModel}. */
+public class LogisticRegressionModelData {
+
+    public final DenseVector coefficient;
+
+    public LogisticRegressionModelData(DenseVector coefficient) {
+        this.coefficient = coefficient;
+    }
+
+    /**
+     * Converts the table model to a data stream.
+     *
+     * @param modelData The table model data.
+     * @return The data stream model data.
+     */
+    public static DataStream<LogisticRegressionModelData> getModelDataStream(Table modelData) {
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) modelData).getTableEnvironment();
+        return tEnv.toDataStream(modelData).map(x -> (LogisticRegressionModelData) x.getField(0));
+    }
+
+    /** Data encoder for {@link LogisticRegressionModel}. */
+    public static class ModelDataEncoder implements Encoder<LogisticRegressionModelData> {
+
+        @Override
+        public void encode(LogisticRegressionModelData modelData, OutputStream outputStream)
+                throws IOException {
+            DenseVectorSerializer serializer = new DenseVectorSerializer();
+            serializer.serialize(
+                    modelData.coefficient, new DataOutputViewStreamWrapper(outputStream));
+        }
+    }
+
+    /** Data decoder for {@link LogisticRegressionModel}. */
+    public static class ModelDataDecoder extends SimpleStreamFormat<LogisticRegressionModelData> {
```

Comment:
For code consistency, could you make change the corresponding class names to `ModelDataDecoder` for `KMeans` and `NaiveBayes` (after the NaiveBayes PR is merged)?

Reply:
`KmeansModelData` is already updated. For naivebayes, how about we do the update in NaiveBayes PR given that it is not merged yet?